# Copyright 2024 Dell Inc. or its subsidiaries. All Rights Reserved.
#
#  Licensed under the Apache License, Version 2.0 (the "License");
#  you may not use this file except in compliance with the License.
#  You may obtain a copy of the License at
#
#      http://www.apache.org/licenses/LICENSE-2.0
#
#  Unless required by applicable law or agreed to in writing, software
#  distributed under the License is distributed on an "AS IS" BASIS,
#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
#  See the License for the specific language governing permissions and
#  limitations under the License.
---

# Usage: prepare_oim_status
inv_directory_path: "/opt/omnia/omnia_inventory"
compute_hostname_ip_filename: "compute_hostname_ip"
inv_file_path_list: "{{ inv_directory_path }}/{{ compute_hostname_ip_filename }}"
xcatd_service: "xcatd.service"
postgresql_service_rhel: "postgresql.service"
postgresql_service_ubuntu: "postgresql"
prepare_oim_execution_req: "Failed. If run the discovery/discovery.yml playbook, please ensure that you run the prepare_oim/prepare_oim.yml playbook before
executing the discovery/discovery.yml playbook. If you're encountering this issue while running the discovery_provision.yml playbook,
please cleanup provision tasks by running the 'ansible-playbook utils/oim_cleanup.yml --tags provision'.
After verifying your input files and Omnia Infrastructure Manager admin NIC configuration, re-run the playbook discovery_provision.yml."
xcat_path: /opt/xcat/bin
oim_os_redhat: "redhat"
oim_os_rocky: "rocky"
oim_os_ubuntu: "ubuntu"
postgresql_start_fail_msg: "Failed to start postgresql services"
xcat_start_fail_msg: "Failed to start xcatd services"
service_retries: 3

# Usage: pre_requisite.yml
provision_config_filename: "{{ role_path }}/../../../input/provision_config.yml"
provision_vault_path: "{{ role_path }}/../../../input/.provision_vault_key"
ansible_vault_search_key: "$ANSIBLE_VAULT;"
provision_config_syntax_fail_msg: "Failed. Syntax errors present in provision_config.yml. Fix errors and re-run playbook again."
provision_credentials_config_filename: "{{ role_path }}/../../../input/provision_config_credentials.yml"
provision_credentials_vault_path: "{{ role_path }}/../../../input/.provision_credential_vault_key"
provision_config_cred_syntax_fail_msg: "Failed. Syntax errors present in provision_credentials_config.yml. Fix errors and re-run playbook again."
conf_file_mode: "0644"
local_repo_config_file: "{{ role_path }}/../../../input/local_repo_config.yml"
local_repo_config_syntax_fail_msg: "Failed. Syntax errors present in local_repo_config.yml. Fix errors and re-run playbook again."

# Usage: Validate_provision_credentials.yml
postgresdb_password_fail_msg: "Failed. postgresdb_password should contain only alphanumeric characters and minimum length 8 in provision_config.yml"
min_length: 8
max_length: 30

# Usage: main.yml
xcat_root_env: "/opt/xcat"
xcat_path_env: "/opt/xcat/bin:/opt/xcat/sbin:/opt/xcat/share/xcat/tools"
xcat_manpath_env: "/opt/xcat/share/man:$MANPATH"
perl_badlang_env: 0
